src/main/java/edu/cornell/kfs/concur/web/struts/ConcurManageAccessTokenAction.java

```diff
@@ -0,0 +1,80 @@
+package edu.cornell.kfs.concur.web.struts;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.struts.action.ActionForm;
+import org.apache.struts.action.ActionForward;
+import org.apache.struts.action.ActionMapping;
+import org.kuali.kfs.kns.question.ConfirmationQuestion;
+import org.kuali.kfs.kns.web.struts.action.KualiAction;
+import org.kuali.kfs.krad.util.GlobalVariables;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.rice.core.api.config.property.ConfigurationService;
+
+import edu.cornell.kfs.concur.ConcurConstants;
+import edu.cornell.kfs.concur.ConcurKeyConstants;
+import edu.cornell.kfs.concur.service.ConcurAccessTokenService;
+
+@SuppressWarnings("deprecation")
+public class ConcurManageAccessTokenAction extends KualiAction {
+
+    public ActionForward start(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) throws Exception {
+        return mapping.findForward(KFSConstants.MAPPING_BASIC);
+    }
+
+    public ActionForward replaceToken(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) throws Exception {
+        Object question = request.getParameter(KFSConstants.QUESTION_INST_ATTRIBUTE_NAME);
+        boolean revokeOldToken;
+        
+        if (ObjectUtils.isNull(question)) {
+            if (getConcurAccessTokenService().currentAccessTokenExists()) {

```

Comment:
The problem is that our current non-PROD databases are being auto-updated with a specific access token (as per our SQL scripts). Thus, if someone validates this in DEV or TEST and chooses to revoke that token, then the Concur jobs in DEV and TEST will stop working upon each refresh.

Would it make sense to have some kind of temporary SQL to alter or remove the DEV/TEST token stored in the database, so that the token from our existing scripts won't get revoked by mistake when validating the change? That would eliminate the need for this dialog.